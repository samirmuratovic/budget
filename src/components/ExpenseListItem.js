import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'

const ExpenseListItem = ({ id, description, amount, note, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount}</p>
    <p>{note}</p>
    <p>{moment(createdAt).calendar()}</p>
  </div>
)

export default ExpenseListItem
