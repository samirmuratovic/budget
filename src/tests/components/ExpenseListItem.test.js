import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import ExpenseListItem from '../../components/ExpenseListItem'

test('should render ExpenseListItem with fixture data', () => {
  const expense = expenses[1]
  const wrapper = shallow(<ExpenseListItem {...expense} key={expense.id} />)

  expect(wrapper).toMatchSnapshot()
})
