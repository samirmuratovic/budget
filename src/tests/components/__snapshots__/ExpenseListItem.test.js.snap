// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with fixture data 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    199965
  </p>
  <p />
  <p>
    12/27/1969
  </p>
</div>
`;
